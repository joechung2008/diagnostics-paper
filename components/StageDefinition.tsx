import { useMemo } from "react";
import { Text, View } from "react-native";
import { useStageDefinitionStyles } from "../hooks/useStageDefinitionStyles";
import type { KeyValuePair, StageDefinitionProps } from "../lib/types";

const StageDefinition = ({ stageDefinition }: StageDefinitionProps) => {
  const dynamicStyles = useStageDefinitionStyles();

  const items = useMemo(
    () =>
      Object.entries(stageDefinition).reduce<KeyValuePair<string>[]>(
        (previous, [key, value]) => [
          ...previous,
          { key, value: value.join(", ") },
        ],
        []
      ),
    [stageDefinition]
  );

  return (
    <View style={dynamicStyles.container}>
      <Text style={dynamicStyles.title}>Stage Definitions</Text>
      <View style={dynamicStyles.table}>
        <View style={dynamicStyles.header}>
          <Text style={dynamicStyles.headerText}>Key</Text>
          <Text style={dynamicStyles.headerText}>Value</Text>
        </View>
        {items.map((item) => (
          <View key={item.key} style={dynamicStyles.row}>
            <Text style={[dynamicStyles.cell, dynamicStyles.cellText]}>
              {item.key}
            </Text>
            <Text style={[dynamicStyles.cell, dynamicStyles.cellText]}>
              {item.value}
            </Text>
          </View>
        ))}
      </View>
    </View>
  );
};

export default StageDefinition;
