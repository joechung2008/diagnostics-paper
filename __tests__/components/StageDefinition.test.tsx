import React from "react";
import { PaperProvider, MD3LightTheme } from "react-native-paper";
import { render } from "@testing-library/react-native";
import StageDefinition from "../../components/StageDefinition";

describe("StageDefinition", () => {
  it("renders correctly with stage definition data", () => {
    const mockStageDefinition = {
      stage1: ["step1", "step2", "step3"],
      stage2: ["step4"],
      stage3: ["step5", "step6"],
    };

    const { toJSON } = render(
      <PaperProvider theme={MD3LightTheme}>
        <StageDefinition stageDefinition={mockStageDefinition} />
      </PaperProvider>
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly with empty stage definition", () => {
    const { toJSON } = render(
      <PaperProvider theme={MD3LightTheme}>
        <StageDefinition stageDefinition={{}} />
      </PaperProvider>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
