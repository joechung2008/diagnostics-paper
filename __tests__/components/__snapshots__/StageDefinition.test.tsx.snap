// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StageDefinition renders correctly with empty stage definition 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFBFE",
    }
  }
>
  <Text
    style={
      {
        "color": "#1C1B1F",
        "fontFamily": "System",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Stage Definitions
  </Text>
  <View
    style={
      {
        "backgroundColor": "#FFFBFE",
        "borderColor": "#79747E",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E7E0EC",
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#1C1B1F",
            "flex": 1,
            "fontFamily": "System",
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Key
      </Text>
      <Text
        style={
          {
            "color": "#1C1B1F",
            "flex": 1,
            "fontFamily": "System",
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
  </View>
</View>
`;

exports[`StageDefinition renders correctly with stage definition data 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFBFE",
    }
  }
>
  <Text
    style={
      {
        "color": "#1C1B1F",
        "fontFamily": "System",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Stage Definitions
  </Text>
  <View
    style={
      {
        "backgroundColor": "#FFFBFE",
        "borderColor": "#79747E",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E7E0EC",
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#1C1B1F",
            "flex": 1,
            "fontFamily": "System",
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Key
      </Text>
      <Text
        style={
          {
            "color": "#1C1B1F",
            "flex": 1,
            "fontFamily": "System",
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        stage1
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        step1, step2, step3
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        stage2
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        step4
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        stage3
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        step5, step6
      </Text>
    </View>
  </View>
</View>
`;
